src/nested_pandas/nestedframe/core.py

```diff
@@ -915,7 +921,27 @@ def reduce(self, func, *args, **kwargs) -> NestedFrame:  # type: ignore[override
                 iterators.append(self[layer].array.iter_field_lists(col))
 
         results = [func(*cols, *extra_args, **kwargs) for cols in zip(*iterators)]
-        return NestedFrame(results, index=self.index)
+        results_nf = NestedFrame(results, index=self.index)
+
+        if infer_nesting:
+            # find potential nested structures from columns
+            nested_cols = []
+            for column in results_nf.columns:
+                if isinstance(column, str) and "." in column:
+                    layer, col = column.split(".", 1)
+                    nested_cols.append(layer)
+            nested_cols = list(np.unique(nested_cols))
```

Comment:
```suggestion
            nested_cols = list(np.unique(
                [column.split(".", 1)[0]
                 for column in results_nf.columns
                 if isinstance(column, str) and "." in column]
            ))
```

Feel free to ignore this one; I can see that it's hardly any fewer lines! I just have preference for list comprehensions.